app/src/main/java/org/covidwatch/android/ui/menu/MenuFragment.kt

```diff
@@ -1,32 +1,39 @@
 package org.covidwatch.android.ui.menu
 
-import android.content.Context
 import android.content.Intent
 import android.net.Uri
 import android.os.Bundle
 import android.view.View
 import android.widget.ImageView
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.LiveData
 import androidx.navigation.fragment.findNavController
 import androidx.recyclerview.widget.DividerItemDecoration
 import androidx.recyclerview.widget.RecyclerView
 import org.covidwatch.android.R
-import kotlinx.coroutines.GlobalScope
-import org.covidwatch.android.data.CovidExposureInformation
+import org.covidwatch.android.data.CovidExposureSummary
 import org.covidwatch.android.data.exposureinformation.ExposureInformationRepository
-import org.covidwatch.android.data.toCovidExposureInformation
-import org.covidwatch.android.exposurenotification.RandomEnObjects
-import org.covidwatch.android.extension.io
+import org.covidwatch.android.data.pref.SharedPreferenceStorage
 import org.koin.android.ext.android.inject
-import org.covidwatch.android.exposurenotification.*
-
+import androidx.lifecycle.Observer
 
 class MenuFragment : Fragment(R.layout.fragment_menu) {
 
     val exposureInformationRepository: ExposureInformationRepository by inject()
 
+    val exposureSummary: LiveData<CovidExposureSummary>
+        get() = SharedPreferenceStorage(requireContext()).observableExposureSummary
```

Comment:
Please, use DI as I was asking before in the previous PRs.
We use Koin: https://insert-koin.io/

Review its documentation to familiarize yourself with it
https://start.insert-koin.io/#/quickstart/android